pkg/rancher-desktop/integrations/manageLinesInFile.ts

```diff
@@ -1,80 +1,135 @@
-  // read file, creating it if it doesn't exist
-  let currentContent: string;
+  const copyFlags = fs.constants.COPYFILE_EXCL | fs.constants.COPYFILE_FICLONE;
+  const desired = getDesiredLines(desiredManagedLines, desiredPresent);
+  let fileStats: fs.Stats;
 
   try {
-    currentContent = await fs.promises.readFile(path, 'utf8');
-  } catch (error: any) {
-    if (error.code === 'ENOENT' && desiredPresent) {
-      const lines = buildFileLines([], desiredManagedLines, ['']);
-      const content = lines.join(os.EOL);
+    fileStats = await fs.promises.lstat(path);
+  } catch (ex: any) {
+    if (ex && 'code' in ex && ex.code === 'ENOENT') {
+      // File does not exist.
+      const content = computeTargetContents('', desired);
 
-      await fs.promises.writeFile(path, content, { mode: DEFAULT_FILE_MODE });
+      if (content) {
+        await fs.promises.writeFile(path, content, { mode: DEFAULT_FILE_MODE });
+      }
 
-      return;
-    } else if (error.code === 'ENOENT' && !desiredPresent) {
       return;
     } else {
-      throw error;
+      throw ex;
     }
   }
 
-  // split file into three parts
-  let before: string[];
-  let currentManagedLines: string[];
-  let after: string[];
+  if (fileStats.isFile()) {
+    if (await fileHasExtendedAttributes(path)) {
+      throw new Error(`Refusing to manage ${ path } which has extended attributes`);
+    }
 
-  try {
-    const currentLines = currentContent.split('\n');
+    const tempName = `${ path }.rd-temp`;
 
-    [before, currentManagedLines, after] = splitLinesByDelimiters(currentLines);
-  } catch (error) {
-    throw new Error(`could not split ${ path }: ${ error }`);
-  }
+    await fs.promises.copyFile(path, tempName, copyFlags);
+
+    try {
+      const currentContents = await fs.promises.readFile(path, 'utf-8');
+      const targetContents = computeTargetContents(currentContents, desired);
+
+      if (targetContents === undefined) {
+        // No changes are needed
+        return;
+      }
+
+      if (targetContents === '') {
+        // The resulting file is empty; unlink it.
+        await fs.promises.unlink(path);
+
+        return;
+      }
 
-  // make the changes
-  if (desiredPresent && !isEqual(currentManagedLines, desiredManagedLines)) {
-    // This is needed to ensure the file ends with an EOL
-    if (after.length === 0) {
-      after = [''];
+      await fs.promises.writeFile(tempName, targetContents, 'utf-8');
+      await fs.promises.rename(tempName, path);
+    } finally {
+      try {
+        await fs.promises.unlink(tempName);
+      } catch {
+        // Ignore errors unlinking the temporary file; if everything went well,
+        // it no longer exists anyway.
+      }
     }
-    const newLines = buildFileLines(before, desiredManagedLines, after);
-    const newContent = newLines.join(os.EOL);
+  } else if (fileStats.isSymbolicLink()) {
+    const backupPath = `${ path }.rd-backup~`;
```

Comment:
Don't we want to check if `backupPath` already exist, and refuse to overwrite it, if it does?

Reply:
`copyFlags` includes `COPYFILE_EXCL`, so if it already exists, we throw.